cmd/node/main.go

```diff
@@ -1734,6 +1741,9 @@ func createNodesCoordinator(
 	if err != nil {
 		return nil, err
 	}
+	if shardIDAsObserver == core.DisabledShardIDAsObserver {
```

Comment:
move this if inside the processDestinationShardAsObserver function.

Reply:
I can't place that logic inside processDestinationShardAsObserver because in epoch bootstrapper I need that information, so don't replace `Disabled` with `0` which you cannot tell if it is default or set